src/xunit.netcore.extensions/Discoverers/ConditionalFactDiscoverer.cs

```diff
@@ -23,45 +24,71 @@ public ConditionalFactDiscoverer(IMessageSink diagnosticMessageSink) : base(diag
+            List<string> falseConditions = new List<string>();
 
-            if (symbols.Length == 2)
+            foreach (string entry in conditionMemberNames)
             {
-                conditionMemberName = symbols[1];
-                ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
-                if (type != null)
+                string conditionMemberName = entry;
+                string[] symbols = conditionMemberName.Split('.');
+                Type declaringType = testMethodDeclaringType;
+
+                if (symbols.Length == 2)
                 {
-                    declaringType = type.ToRuntimeType();
+                    conditionMemberName = symbols[1];
+                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
+                    if (type != null)
+                    {
+                        declaringType = type.ToRuntimeType();
+                    }
                 }
-            }
-            
-            MethodInfo conditionMethodInfo;
-            if (conditionMemberName == null ||
-                (conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
-            {
-                return new[] {
+
+                MethodInfo conditionMethodInfo;
+                if (conditionMemberName == null ||
+                    (conditionMethodInfo = LookupConditionalMethod(declaringType, conditionMemberName)) == null)
+                {
+                    return new[] {
                     new ExecutionErrorTestCase(
-                        _diagnosticMessageSink,
+                        diagnosticMessageSink,
                         discoveryOptions.MethodDisplayOrDefault(),
                         testMethod,
                         GetFailedLookupString(conditionMemberName))
                 };
-            }
+                }
 
-            IEnumerable<IXunitTestCase> testCases = base.Discover(discoveryOptions, testMethod, factAttribute);
-            if ((bool)conditionMethodInfo.Invoke(null, null))
-            {
-                return testCases;
+                // In the case of multiple conditions, collect the results of all
+                // of them to produce a summary skip reason.
+                if (!(bool)conditionMethodInfo.Invoke(null, null))
+                {
+                    falseConditions.Add(conditionMemberName);
+                }
             }
-            else
+
+            // Compose a summary of all conditions that returned false.
+            if (falseConditions.Count > 0)
             {
-                string skippedReason = "\"" + conditionMemberName + "\" returned false.";
+                string skippedReason = String.Format("Condition(s) not met: \"{0}\"", String.Join("\", \"", falseConditions));
```

Comment:
We use `string.Format` vs `String.Format`.  It feels wrong to me, too, but it's in [the rules](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) (number 11).